core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/ChunkBuffer.java

```diff
@@ -88,6 +93,33 @@ public synchronized Iterable<ByteBuffer> split(ByteBuffer inputByteBuffer) {
         return byteBuffers;
     }
 
+    private synchronized Iterable<ByteBuffer> splitWithUnknownLength(ByteBuffer inputByteBuffer) {
+        boolean isLastChunk = inputByteBuffer.remaining() != chunkSize;
+        List<ByteBuffer> bufferedList = new ArrayList<>();
+
+        while (inputByteBuffer.hasRemaining()) {
+            int bytesToCopy = Math.min(inputByteBuffer.remaining(), currentBuffer.remaining());
+            byte[] bytes = new byte[bytesToCopy];
```

Comment:
Can we avoid copying the data to another byte array? See fillCurrentBuffer

Reply:
removed this method, not needed anymore